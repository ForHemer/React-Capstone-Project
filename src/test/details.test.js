import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import AnimalDetail from '../components/animal/details';

describe('Testing Animal Details', () => {
  it('Render the Animal Details page, should match to the snapshot', () => {
    const list = renderer
      .create(
        <Provider store={store}>
          <Router>
            <AnimalDetail />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(list).toMatchSnapshot();
  });
});
