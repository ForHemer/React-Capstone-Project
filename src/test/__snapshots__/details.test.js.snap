// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Animal Details Render the Animal Details page, should match to the snapshot 1`] = `
<div
  className="detailStyle row"
>
  <a
    className="animalDetailLink"
    href="/"
    onClick={[Function]}
  >
    <button
      className="buttonLeft"
      onClick={[Function]}
      type="button"
    >
      Back
    </button>
  </a>
  <ul
    className="liste"
  >
    <div
      className="detailStyle list-group-item"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b5b98",
        }
      }
    >
      <img
        className="card-img"
        style={
          Object {
            "width": "150px",
          }
        }
      />
    </div>
    <div
      className="list-group-item"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b5b98",
          "color": "#fff",
        }
      }
    >
      <span>
        Name : 
      </span>
      <span />
    </div>
    <div
      className="list-group-item"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b5b98",
          "color": "#fff",
        }
      }
    >
      <span>
        Type : 
      </span>
      <span />
    </div>
    <div
      className="list-group-item"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b5b98",
          "color": "#fff",
        }
      }
    >
      <span>
        Latin Name : 
      </span>
      <span />
    </div>
    <div
      className="list-group-item"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b5b98",
          "color": "#fff",
        }
      }
    >
      <span>
        Active time : 
      </span>
      <span />
    </div>
    <div
      className="list-group-item"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b5b98",
          "color": "#fff",
        }
      }
    >
      <span>
        Length (min) : 
      </span>
      <span />
    </div>
    <div
      className="list-group-item"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b5b98",
          "color": "#fff",
        }
      }
    >
      <span>
        Length (max) : 
      </span>
      <span />
    </div>
    <div
      className="list-group-item"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b5b98",
          "color": "#fff",
        }
      }
    >
      <span>
        Weight (min) : 
      </span>
      <span />
    </div>
    <div
      className="list-group-item"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b5b98",
          "color": "#fff",
        }
      }
    >
      <span>
        Weight (max) : 
      </span>
      <span />
    </div>
    <div
      className="list-group-item"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b5b98",
          "color": "#fff",
        }
      }
    >
      <span>
        Lifespan : 
      </span>
      <span />
    </div>
    <div
      className="list-group-item"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b5b98",
          "color": "#fff",
        }
      }
    >
      <span>
        Habitat : 
      </span>
      <span />
    </div>
    <div
      className="list-group-item"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b5b98",
          "color": "#fff",
        }
      }
    >
      <span>
        Diet : 
      </span>
      <span />
    </div>
    <div
      className="list-group-item"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b5b98",
          "color": "#fff",
        }
      }
    >
      <span>
        Geography : 
      </span>
      <span />
    </div>
  </ul>
</div>
`;
